libs/platform/src/lib/components/upload-collection/dialogs/move-to/move-to.component.ts

```diff
@@ -0,0 +1,125 @@
+import { Component, Inject, OnInit } from '@angular/core';
+
+import { DIALOG_REF, DialogRef, DialogService, DialogConfig } from '@fundamental-ngx/core';
+
+import { UploadCollectionItem, UploadCollectionFolder } from '../../models/upload-collection.models';
+import { NewFolderComponent } from '../new-folder';
+
+function isObject<T>(item: T): boolean {
+    return typeof item === 'object' && !Array.isArray(item) && item !== null;
+}
+
+@Component({
+    templateUrl: './move-to.component.html',
+    styles: [`
+        .fd-list__item--alternative-background {
+            background: #f2f2f2;
+        }
+
+        .fd-list__icon--alternative-background {
+            color: #74777A !important;
+        }
+
+        .fd-dialog__body--no-horizontal-padding {
+            padding-left: 0 !important;
+            padding-right: 0 !important;
+        }
+    `]
+})
+export class MoveToComponent implements OnInit {
+    readonly _items: UploadCollectionItem[] = this.dialogRef.data.items;
+    _foldersList: UploadCollectionFolder[] = [];
+    _currentFolder?: UploadCollectionFolder = this.dialogRef.data.currentFolder;
+    selectedFolder?: UploadCollectionFolder;
+
+    constructor(
+        @Inject(DIALOG_REF) private readonly dialogRef: DialogRef,
+        private readonly _dialogService: DialogService
+    ) {}
+
+    ngOnInit(): void {
+        this._init(this._currentFolder ? this._currentFolder.files : this._items);
+    }
+
+    goToParentFolder(): void {
+        this.selectedFolder = null;
+        this._currentFolder = this._getParentFolderByCurrentFolderId(this._currentFolder.documentId);
+        this._init(this._currentFolder ? this._currentFolder.files : this._items);
+    }
+
+    close(): void {
+        this.dialogRef.dismiss();
+    }
+
+    confirm(): void {
+        this.dialogRef.close({
+            selectedFolder: this.selectedFolder || this._currentFolder
+        });
+    }
+
+    openFolder(folder: UploadCollectionFolder): void {
+        this.selectedFolder = null;
+        this._currentFolder = folder;
+        this._init(folder.files);
+    }
+
+    selectFolder(folder: UploadCollectionFolder): void {
+        this.selectedFolder = folder;
+    }
+
+    hasFolders(items: UploadCollectionItem[]): boolean {
+        return items.some((item) => item.type === 'folder');
+    }
+
+    newFolder(): void {
+        const dialogRef = this._dialogService.open(NewFolderComponent, {
+            responsivePadding: true,
+            backdropClickCloseable: false,
+            height: '70%',
+            backdropClass: 'fdp-upload-collection-dialig-no-bg',
+            data: {
+                currentFolder: this._currentFolder
+            }
+        } as DialogConfig);
+
+        dialogRef.afterClosed.subscribe((folderName) => {
```

Comment:
Create a reference to this subscription and `unsubscribe` in OnDestroy lifecycle hook for this component.

Reply:
fixed